libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -94,6 +94,15 @@ void AP_Logger_File::Init()
         _log_directory = custom_dir;
     }
 
+    uint16_t last_log_num = find_last_log();
+    if (last_log_is_marked_discard) {
+        // delete the last log leftover from LOG_DISARMED=3
+        char *filename = _log_file_name(last_log_num);
```

Comment:
We're leaking this; comment is pretty clear on the method :-)

Reply:
thanks! fixed